test/Makefile

```diff
@@ -18,21 +18,14 @@ clean:
 
 bins: $(BINS)
 
-bin/random: $(RANDOM_SRC)/**/*.go IPFS-BUILD-OPTIONS
-	@echo "*** installing $@ ***"
-	go build $(GOFLAGS) -o bin/random $(RANDOM_SRC)/random
-
-bin/multihash: $(MULTIHASH_SRC)/**/*.go IPFS-BUILD-OPTIONS
-	@echo "*** installing $@ ***"
-	go build $(GOFLAGS) -o bin/multihash $(MULTIHASH_SRC)/multihash
-
-bin/ipfs: $(IPFS_ROOT)/**/*.go IPFS-BUILD-OPTIONS
-	@echo "*** installing $@ ***"
-	go build $(GOFLAGS) -o bin/ipfs $(IPFS_CMD)
-
-bin/pollEndpoint: $(POLLENDPOINT_SRC)/*.go IPFS-BUILD-OPTIONS
-	@echo "*** installing $@ ***"
-	go build $(GOFLAGS) -o bin/pollEndpoint $(POLLENDPOINT_SRC)
+bin/random: BIN_SOURCES = $(RANDOM_SRC)/random
+bin/multihash: BIN_SOURCES = $(MULTIHASH_SRC)/multihash
+bin/ipfs: BIN_SOURCES = $(IPFS_CMD)
+bin/pollEndpoint: BIN_SOURCES = $(POLLENDPOINT_SRC)
+
+bin/%: FORCE IPFS-BUILD-OPTIONS
+	@echo "*** building and installing test local version of '$*' binary ***"
```

Comment:
i think the shorter message is fine. 

if the goal is to document what's going on here, maybe we can add a note to the readme or the makefile itself.